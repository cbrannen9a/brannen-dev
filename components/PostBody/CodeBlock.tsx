import React, { FC } from "react";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";

const CodeBlock: FC<Props> = ({ language = "javascript", children }) => {
  return <SyntaxHighlighter language={language}>{children}</SyntaxHighlighter>;
};

interface Props {
  language: string;
}

export default CodeBlock;
