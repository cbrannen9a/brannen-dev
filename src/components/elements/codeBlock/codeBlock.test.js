import React from 'react';
import renderer from 'react-test-renderer';

import CodeBlock from './codeBlock';

const code =
    "import test from 'tester';\n\nconst thistest = () => {\n  return <div>testing</div>\n}";
const language = 'jsx';

describe('CodeBlock', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<CodeBlock language={language}>{code}</CodeBlock>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
