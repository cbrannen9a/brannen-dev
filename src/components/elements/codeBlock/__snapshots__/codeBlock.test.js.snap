// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock renders correctly 1`] = `
<pre
  className="language-javascript"
>
  <code
    className="language-javascript"
  >
    import test from 'tester';

const thistest = () =&gt; {
  return &lt;div&gt;testing&lt;/div&gt;
}
  </code>
</pre>
`;
